### PR TITLE
crosscluster: redact url for LDR jobs

### PR DESCRIPTION
**Pull request by azhu-crl**
Redact sensitive information from URLs in job descriptions. This PR also adds a `description` column with redacted URLs to rows returned by `SHOW LOGICAL REPLICATION JOBS WITH DETAILS`

Example output of updated `SHOW LOGICAL REPLICATION JOBS WITH DETAILS` command:
```
demo@127.0.0.1:26257/demoapp/b> SHOW LOGICAL REPLICATION JOBS WITH DETAILS;
        job_id       | status  | targets | replicated_time |    replication_start_time     | conflict_resolution_type |                                                                                           description
---------------------+---------+---------+-----------------+-------------------------------+--------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  998275879939801089 | running | {tab1}  | NULL            | 2024-08-27 00:53:41.560452+00 | DLQ                      | LOGICAL REPLICATION STREAM into a.public.tab1 from postgresql://redacted@127.0.0.1:26257/b?options=-ccluster%3Ddemoapp&sslmode=require&sslrootcert=%2FUsers%2Fannezhu%2F.cockroach-demo%2Fca.crt
  998275879987052545 | running | {tab1}  | NULL            | 2024-08-27 00:53:41.575131+00 | LWW                      | LOGICAL REPLICATION STREAM into b.public.tab1 from postgresql://redacted@127.0.0.1:26257/a?options=-ccluster%3Ddemoapp&sslmode=require&sslrootcert=%2FUsers%2Fannezhu%2F.cockroach-demo%2Fca.crt
(2 rows)

Time: 13ms total (execution 12ms / network 0ms)
```

Epic: none
Informs: https://github.com/cockroachdb/cockroach/issues/128960
Release note: none

### CODE REVIEW

File: pkg/ccl/crosscluster/logical/logical_replication_job_test.go [hunk 1]
```diff
@@ -1456,3 +1467,64 @@ func TestShowLogicalReplicationJobs(t *testing.T) {
...
+
+	server, s, dbA, dbB := setupLogicalTestServer(t, ctx, testClusterBaseClusterArgs, 1)
+	defer server.Stopper().Stop(ctx)
+
+	dbAURL, cleanup := s.PGUrl(t, serverutils.DBName("a"))
+	defer cleanup()
+	dbBURL, cleanupB := s.PGUrl(t, serverutils.DBName("b"))
+	defer cleanupB()
+
+	redactedDbAURL := strings.Replace(dbAURL.String(), username.RootUser, `redacted`, 1)
```
Comment (msbutler) [inline pkg/ccl/crosscluster/logical/logical_replication_job_test.go:1484]: could the redacted URL get tested in  TestShowLogicalReplicationJobs ? So, we don't need to create a whole new test for this?

**Review by msbutler — APPROVED** [review]
LGTM! there are two logical changes in this PR: 1) redact the url in the description; 2) add the description to SHOW LOGICAL REPLICATION JOBS. Could you split this PR into two commits to reflect this?

File: pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go [hunk 2]
```diff
@@ -195,6 +197,15 @@ func createLogicalReplicationStreamPlanHook(
 		}
 		streamAddress = crosscluster.StreamAddress(streamURL.String())
 
+		cleanedURI, err := cloud.SanitizeExternalStorageURI(from, nil)
+		if err != nil {
+			return err
+		}
+		cleanedURI, err = changefeedbase.RedactUserFromURI(cleanedURI)
```
Comment (dt) [inline pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:204]: Do we need `RedactUserFromURI` ? We're redacting the password in `SanitizeExternalStorageURI` but user seems useful, no?

File: pkg/ccl/crosscluster/logical/logical_replication_job_test.go [hunk 3]
```diff
@@ -1456,3 +1456,59 @@ func TestShowLogicalReplicationJobs(t *testing.T) {
 	jobutils.WaitForJobToCancel(t, dbA, jobAID)
 	jobutils.WaitForJobToCancel(t, dbA, jobBID)
 }
+
+func TestLogicalReplicationJobDescription(t *testing.T) {
```
Comment (msbutler) [inline pkg/ccl/crosscluster/logical/logical_replication_job_test.go:1471]: we spoke offline: i think we decided to delete this test?

**Review by msbutler — APPROVED** [review]

File: pkg/ccl/crosscluster/logical/logical_replication_job_test.go (hunk 3 above)
Comment (azhu-crl) [reply pkg/ccl/crosscluster/logical/logical_replication_job_test.go:1471]: Yep! I'll push another update shortly

**Comment by azhu-crl** [comment]
I removed `TestLogicalReplicationJobDescription` as we discussed. TFTR!

bors r+

**Pull request merged**